src/app/code-set/code-set-details/code-set-details.component.ts

```diff
@@ -38,225 +29,135 @@ import { BroadcastService } from '@mdm/services/broadcast.service';
-  exportList = [];
   addedToFavourite = false;
-  download: any;
-  downloadLink: any;
-  urlText: any;
   currentBranch = '';
   branchGraph = [];
-
-  canEditDescription = true;
-  showEditDescription = false;
-
-  constructor(private resourcesService: MdmResourcesService,
-              private messageService: MessageService,
-              private messageHandler: MessageHandlerService,
-              private securityHandler: SecurityHandlerService,
-              private stateHandler: StateHandlerService,
-              private sharedService: SharedService,
-              private elementDialogueService: ElementSelectorDialogueService,
-              private broadcastSvc: BroadcastService,
-              private dialog: MatDialog,
-              private favouriteHandler: FavouriteHandlerService,
-              private title: Title,
-              private editingService: EditingService) {
+  access: any;
+
+  constructor(
+    private resourcesService: MdmResourcesService,
+    private validatorService: ValidatorService,
+    private messageService: MessageService,
+    private messageHandler: MessageHandlerService,
+    private securityHandler: SecurityHandlerService,
+    private stateHandler: StateHandlerService,
+    private sharedService: SharedService,
+    private elementDialogueService: ElementSelectorDialogueService,
+    private broadcastSvc: BroadcastService,
+    private dialog: MatDialog,
+    private favouriteHandler: FavouriteHandlerService,
+    private title: Title,
+    private editingService: EditingService
+  ) {
     this.isAdminUser = this.sharedService.isAdmin;
-    this.isLoggedIn = this.securityHandler.isLoggedIn();
-    this.CodeSetDetails();
+     this.codeSetDetails();
   }
 
-  public showAddElementToMarkdown() { // Remove from here & put in markdown
+  public showAddElementToMarkdown() {
+    // Remove from here & put in markdown
     this.elementDialogueService.open('Search_Help', 'left' as DialogPosition);
   }
 
   ngOnInit() {
-    this.editableForm = new EditableDataModel();
-    this.editableForm.visible = false;
-    this.editableForm.deletePending = false;
-
-
-    this.editableForm.show = () => {
-      this.editForm.forEach(x => x.edit({
-        editing: true,
-        focus: x.name === 'moduleName' ? true : false
-      }));
-      this.editableForm.visible = true;
-    };
-
-    this.editableForm.cancel = () => {
-      this.editingService.stop();
-      this.editForm.forEach(x => x.edit({ editing: false }));
-      this.editableForm.visible = false;
-      this.editableForm.validationError = false;
-      this.errorMessage = '';
-      this.editableForm.description = this.result.description;
-      if (this.result.classifiers) {
-        this.result.classifiers.forEach(item => {
-          this.editableForm.classifiers.push(item);
-        });
+    this.subscription = this.messageService.changeUserGroupAccess.subscribe(
+      (message: boolean) => {
+        this.showSecuritySection = message;
```

Comment:
This might need changing, using a subscription service is probably overkill. My idea was to have the parent component load the detail/access properties and pass them down to the child component via `@Input()`, if changes need detecting then maybe use `@Output()` and/or `ngOnChanges()`.

Defer this to a new clean up issue, just writing down as a reminder.